src/commands/periscope/periscope.ts

```diff
@@ -0,0 +1,248 @@
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as buffer from 'buffer';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import { IActionContext } from 'vscode-azureextensionui';
+import { parseResource } from '../../azure-api-utils';
+import * as tmpfile from '../utils/tempfile';
+import { getClusterKubeconfig } from '../../extension';
+import { getExtensionPath, longRunning } from '../utils/host';
+import {
+  getClusterDiagnosticSettings,
+  getStorageAccountKeyUI,
+  getStorageInfo,
+  replaceDeploymentAccountNameAndSas,
+  generateDownloadableLinks,
+  getWebviewContent
+} from './helpers/storageandhtmlhelper';
+import { PeriscopeStorage } from './models/storage';
+import * as amon from 'azure-arm-monitor';
+
+export default async function periscope(
+  context: IActionContext,
+  target: any
+): Promise<void> {
+  const kubectl = await k8s.extension.kubectl.v1;
+  const cloudExp = await k8s.extension.cloudExplorer.v1;
+
+  if (cloudExp.available && kubectl.available) {
+    const cloudTarget = cloudExp.api.resolveCommandTarget(target);
+    if (cloudTarget && cloudTarget.cloudName === "Azure" &&
+      cloudTarget.nodeType === "resource" && cloudTarget.cloudResource.nodeType === "cluster") {
+      const cloudResource = cloudTarget.cloudResource;
+      const clusterKubeConfig = await getClusterKubeconfig(cloudResource);
+      if (clusterKubeConfig) {
+        /**
+          Three key parts for running aks-periscope on a cluster.
+            1. Get the cluster diagnostic settings for account-storage or get it manually via UI.
+            2. Generate SAS for the storage account selected\provided or default supplied.
+            3. Run the clean-up and then aks-persicope on the cluster with current cluster kubeconfig.
+            4. Generate Dowloadable link once action is performed at the webView.
+        **/
+        await longRunning(`Checking storage associated cluster.`,
+          async () => {
+            // Check if Diagnostic setting exist.
+            const diagnosticSettings = await getClusterDiagnosticSettings(cloudResource);
+
+            if (diagnosticSettings?.value?.length) {
```

Comment:
I don't think this lot should be under the "Checking storage..." progress indicator.  It looks like the idea is to check the storage (long running action 1), then optionally do some UI interaction, then install Periscope (long running action 2).

Remember that `longRunning` (at least if you copied my code) returns the value of the long-running task.  So you can do something like:

```
const diagnosticSettings = await longRunning("...", () =>
    getClusterDiagnosticSettings(cloudResource)
);
```